### PR TITLE
(5) Add `reset()` to thread local ContextVar and no-op `copy_context()`

### PR DESCRIPTION
**Pull request by antonpirker**
Improves the capabilities of our threadlocal context variables, we use when no "real" context variables are available (for example in old Python versions)

- Adds a no-op [copy_context](https://docs.python.org/3/library/contextvars.html#contextvars.copy_context) function to use in environments without context vars
- Adds [reset](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar.reset) functionality to our threadlocal based context vars.

This is preparation work for refactoring how we deal with Hubs and Scopes in the future.

### CODE REVIEW

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -1275,28 +1298,28 @@ def _get_contextvars():
...
+                return True, ContextVar, copy_context
             except ImportError:
                 pass
 
     # Fall back to basic thread-local usage.
 
     from threading import local
 
-    return False, _make_threadlocal_contextvars(local)
+    return False, _make_threadlocal_contextvars(local), _make_noop_copy_context()
```
Comment (50-Course) [inline sentry_sdk/utils.py:1319]: Looks good to me.

**Comment by 50-Course** [comment]
Following this MR 👍🏼 looking forward to the updates

File: sentry_sdk/utils.py [hunk 2]
```diff
@@ -1248,24 +1249,49 @@ def _make_threadlocal_contextvars(local):
...
+            token = str(random.getrandbits(64))
+            original_value = self.get()
+            setattr(self._original_local, token, original_value)
             self._local.value = value
+            return token
+
+        def reset(self, token):
+            # type: (Any) -> None
+            self._local.value = getattr(self._original_local, token)
+            setattr(self._original_local, token, None)
```
Comment (sl0thentr0py) [inline sentry_sdk/utils.py:1274]: isn't it better to delete here so that it gets GCd?

**Review by sl0thentr0py — APPROVED** [review]
I don't yet know how you will use `copy_context` but this change is fine self-contained

File: sentry_sdk/utils.py (hunk 2 above)
Comment (antonpirker) [reply sentry_sdk/utils.py]: @50-Course If it is not garbage collected it stays in memory. With a lot sets and resets the memory usage could grow without memory being freed.

Comment (50-Course) [reply sentry_sdk/utils.py]: Oh okay got it!

**Pull request merged**